Rubberduck.VBEEditor/Events/VBENativeServices.cs

```diff
@@ -165,19 +166,20 @@ private static void OnSelectionChanged(IntPtr hwnd)
             if (pane != null) SelectionChanged?.Invoke(_vbe, new SelectionChangedEventArgs(pane));
         }
 
-        private static string _currentLine;
-        public static event EventHandler<AutoCompleteEventArgs> CaretHidden; // not CodeChanged because wouldn't fire on paste
-        private static void OnCaretHidden(IntPtr hwnd)
+        //private static string _currentLine;
+        public static event EventHandler<AutoCompleteEventArgs> KeyDown; 
+        private static void OnKeyDown(KeyPressEventArgs e)
         {
-            var pane = GetCodePaneFromHwnd(hwnd);
-            if (pane?.Selection.IsSingleCharacter ?? false)
+            using (var pane = GetCodePaneFromHwnd(e.Hwnd))
+            using (var module = pane.CodeModule)
             {
-                var args = new AutoCompleteEventArgs(pane);
-                if (_currentLine != args.OldCode && !string.IsNullOrEmpty(args.OldCode))
-                {
-                    CaretHidden?.Invoke(_vbe, args);
-                    _currentLine = args.NewCode;
-                }
+                var args = new AutoCompleteEventArgs(module, e);
+                //if (_currentLine != args.CurrentLine)
+                //{
+                    KeyDown?.Invoke(_vbe, args);
+                    //_currentLine = args.NewCode;
+                    e.Handled = args.Handled;
+                //}
```

Comment:
Do we still need these commented out lines?

Reply:
We don't; will remove.